### PR TITLE
Add PostgresConfig.ReadTimeout and PostgresConfig.WriteTimeout

### PR DESCRIPTION
**Pull request by SimonXming**
* add `PostgresConfig.ReadTimeout`
* add `PostgresConfig.WriteTimeout`

### CODE REVIEW

File: postgres_config.go [hunk 1]
```diff
@@ -9,6 +9,12 @@ import (
 	"time"
 )
 
+type DBConfig interface {
```
Comment (vito) [inline postgres_config.go]: I don't think this interface and the `GetX` methods are necessary, or at least I don't see an obvious reason to have it exported by the `flag` package.

**Review by vito — CHANGES_REQUESTED** [review]
Thanks! Left some feedback, I'd like to avoid exporting an interface and getters here as it doesn't seem super useful within the package.

File: postgres_config.go (hunk 1 above)
Comment (SimonXming) [reply postgres_config.go]: OK, I thought the interface will be useful if we gonna support multiple database driver(like `mysql`, `sqlite3`). What do you think? 

Maybe it's better if I revert back to https://github.com/concourse/flag/pull/2/commits/d848ba7bbbcdb689a466d8f863c4d09a8f3d73ee.

Comment (vito) [reply postgres_config.go]: Ah, I guess in that situation it makes sense, but we don't plan on doing that any time soon so right now it looks more like [this gotcha](https://github.com/concourse/concourse/wiki/Concourse-Go-Style-Guide#a-public-interface-for-the-sake-of-a-mockable-interface) from our style-guide.

I would prefer removing it for now. :slightly_smiling_face:

Comment (SimonXming) [reply postgres_config.go]: Sure. I will remove it.

File: postgres_config_test.go [hunk 2]
```diff
@@ -23,4 +24,22 @@ var _ = Describe("PostgresConfig", func() {
 			}.ConnectionString()).To(Equal("dbname='atc' host='1.2.3.4' password='password \\\\ with \\' funny ! chars' port=5432 sslmode='verify-full' user='some user'"))
 		})
 	})
+	Describe("Timeout", func() {
```
Comment (vito) [inline postgres_config_test.go]: don't think this test is really necessary, there's no behavior actually added or exercised here

**Review by vito — CHANGES_REQUESTED** [review]
I don't think we need the tests, but they do point out something odd in that there's nothing really to do in this package with these flags.

Maybe they should just be on the ATC instead? The rest of the flags are actually needed for building the connection string, but these flags are only here to be read out in the ATC. It might be kind of misleading to have them here because it's still the caller's responsibility to do something with them.

Maybe they could just be added here?:

https://github.com/concourse/concourse/blob/master/atc/atccmd/command.go#L92

:man_shrugging:

File: postgres_config_test.go (hunk 2 above)
Comment (SimonXming) [reply postgres_config_test.go]: Sure, I already added test cases under `atc/atccmd`.

**Comment by vito** [comment]
@SimonXming Sorry for the confusion, my concern wasn't over where the tests are, but whether we should have the flags here in the first place vs. just adding flags to the ATC, since they're only used there.

I don't think the tests are necessary in either repository, since they don't really cover anything important or risky.

Having the flags here is risky because something might use the whole config type but then never actually respect the read/write timeout flags. I don't think flags should be added to this repo unless they're actually used within this repo (like the other flags on `PostgresConfig`, which are used to build the connection string).

Would it make sense to just abandon this PR and add the flags directly to the `atccmd.Command` struct alongside the `flag.PostgresConfig`?

```go
Postgres             flag.PostgresConfig `group:"PostgreSQL Configuration" namespace:"postgres"`
PostgresReadTimeout  time.Duration       `long:"postgres-read-timeout" ...`
PostgresWriteTimeout time.Duration       `long:"postgres-write-timeout" ...`
```

That would also make merging easier since there wouldn't be two inter-dependent PRs.

**Comment by SimonXming** [comment]
> ```go
> Postgres             flag.PostgresConfig `group:"PostgreSQL Configuration" namespace:"postgres"`
> PostgresReadTimeout  time.Duration       `long:"postgres-read-timeout" ...`
> PostgresWriteTimeout time.Duration       `long:"postgres-write-timeout" ...`
> ```
> 
> That would also make merging easier since there wouldn't be two inter-dependent PRs.

Yeah, `postgres-read-timeout` and `postgres-write-timeout` won't work because `flag.PostgresConfig` already take namespace `postgres-*`. So I'm gonna go with `sql-read-timeout` and `sql-write-timeout`.

**Comment by vito** [comment]
Ah, dang, didn't realize it checked for conflicts like that.

That's OK, I guess `sql-*` might make more sense anyway since it's not really Postgres specific.

**Pull request closed without merging**